server/src/internalClusterTest/java/org/opensearch/remotestore/RemoteSegmentStatsFromNodesStatsIT.java

```diff
@@ -0,0 +1,129 @@
+
+@OpenSearchIntegTestCase.ClusterScope(scope = OpenSearchIntegTestCase.Scope.TEST, numDataNodes = 0)
+public class RemoteSegmentStatsFromNodesStatsIT extends RemoteStoreBaseIntegTestCase {
+    private static final String INDEX_NAME = "remote-index-1";
+    private static final int DATA_NODE_COUNT = 2;
+    private static final int CLUSTER_MANAGER_NODE_COUNT = 3;
+
+    @Before
+    public void setup() {
+        setupCustomCluster();
+        setupRepo(false);
+    }
+
+    private void setupCustomCluster() {
+        internalCluster().startClusterManagerOnlyNodes(CLUSTER_MANAGER_NODE_COUNT);
+        internalCluster().startDataOnlyNodes(DATA_NODE_COUNT);
+        ensureStableCluster(DATA_NODE_COUNT + CLUSTER_MANAGER_NODE_COUNT);
+    }
+
+    public void testNodesStatsParity() {
+        String[] dataNodes = internalCluster().getDataNodeNames().toArray(String[]::new);
+        String randomDataNode = dataNodes[randomIntBetween(0, dataNodes.length - 1)];
+        String firstIndex = INDEX_NAME + "1";
+        String secondIndex = INDEX_NAME + "2";
+
+        // Create first index
+        createIndex(
+            firstIndex,
+            Settings.builder().put(remoteStoreIndexSettings(0, 1)).put("index.routing.allocation.require._name", randomDataNode).build()
+        );
+        ensureGreen(firstIndex);
+        indexSingleDoc(firstIndex);
+        refresh(firstIndex);
+
+        // Create second index
+        createIndex(
+            secondIndex,
+            Settings.builder().put(remoteStoreIndexSettings(0, 1)).put("index.routing.allocation.require._name", randomDataNode).build()
+        );
+        ensureGreen(secondIndex);
+        indexSingleDoc(secondIndex);
+        refresh(secondIndex);
+
+        long cumulativeUploads = 0;
+
+        // Fetch upload stats
+        RemoteStoreStatsResponse remoteStoreStatsFirstIndex = client(randomDataNode).admin()
+            .cluster()
+            .prepareRemoteStoreStats(firstIndex, "0")
+            .setLocal(true)
+            .get();
+        cumulativeUploads += remoteStoreStatsFirstIndex.getRemoteStoreStats()[0].getStats().uploadBytesSucceeded;
+        RemoteStoreStatsResponse remoteStoreStatsSecondIndex = client(randomDataNode).admin()
+            .cluster()
+            .prepareRemoteStoreStats(firstIndex, "0")
+            .setLocal(true)
+            .get();
+        cumulativeUploads += remoteStoreStatsSecondIndex.getRemoteStoreStats()[0].getStats().uploadBytesSucceeded;
+
+        // Fetch nodes stats
+        NodesStatsResponse nodesStatsResponse = client().admin()
+            .cluster()
+            .prepareNodesStats(randomDataNode)
+            .setIndices(new CommonStatsFlags().set(CommonStatsFlags.Flag.Segments, true))
+            .get();
+        RemoteSegmentStats remoteSegmentStats = nodesStatsResponse.getNodes().get(0).getIndices().getSegments().getRemoteSegmentStats();
+        assertEquals(cumulativeUploads, remoteSegmentStats.getUploadBytesSucceeded());
```

Comment:
Why aren't we testing remaining stats? We seem to be testing only `UploadBytesSucceeded`.

Reply:
Added a couple of more ITs to make the suite more exhaustive